sdk/iot/Azure.Iot.Hub.Service/src/DevicesClient.cs

```diff
@@ -43,8 +43,11 @@ internal DevicesClient(RegistryManagerRestClient registryManagerClient, TwinRest
         /// <summary>
         /// Create or update a device identity.
         /// </summary>
-        /// <param name="deviceIdentity">the device identity to create.</param>
-        /// <param name="precondition">The condition on which to perform this operation. To create a device identity, this value must be equal to <see cref="IfMatchPrecondition.Unconditional"/>.</param>
+        /// <param name="deviceIdentity">the device identity to create or update.</param>
+        /// <param name="precondition">The condition on which to perform this operation.
+        /// In case of create, the condition must be equal to <see cref="IfMatchPrecondition.IfMatch"/>.
```

Comment:
...and `DeviceIdentity.Etag` should not be set.

Reply:
I will leave that one out since we will make it readonly.